homeassistant/components/kodi/media_player.py

```diff
@@ -644,6 +677,31 @@ async def async_play_media(self, media_type, media_id, **kwargs):
             await self._kodi.play_playlist(int(media_id))
         elif media_type_lower == "directory":
             await self._kodi.play_directory(str(media_id))
+        elif media_type_lower in [
+            MEDIA_TYPE_ARTIST,
+            MEDIA_TYPE_ALBUM,
+        ]:
+            await self._kodi.clear_playlist()
+            params = {"playlistid": 0, "item": {f"{media_type}id": int(media_id)}}
+            # pylint: disable=protected-access
+            await self._kodi._server.Playlist.Add(params)
```

Comment:
Let's change the library to add support for what we need.

Reply:
Exactly. That's what my PM was about. I'll do a PR for that ASAP. Wanted to get that out though. It definitely needs quite a bit refactoring.